lib/spack/spack/main.py

```diff
@@ -643,10 +665,22 @@ def shell_set(var, value):
         path = spack.util.path.canonicalize_path(path)
```

Comment:
I think that by default we should not set `config:module_roots` (in `etc/defaults/config.yaml`). When using the `install_trees` config (vs. the old `install_tree` config), the first path in the module lookup should be the module root associated with the selected install tree.

Use cases:

* We want to allow users to continue using (updating and reading from) the "top-level" `module_roots:` config if they are using the (now-deprecated) top-level `install_tree` config.
* We want to seamlessly transition to using the install tree modules if they user selects an install tree

It occurs to me that

* we may want to support customizing the module roots on a per-install-tree basis (since admins may want to keep writing to a well-known location)
* we may want to raise an error if the config contains `install_trees` and `module_roots` at the top level like:

```
config:
  # This is the path to the root of the Spack install tree.
  # You can use $spack here to refer to the root of the spack instance.
  install_trees:
    spack-root:
      root: $spack/opt/spack
      projections:
        all: "${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}"
      permissions:
        read: world
        write: user
    user:
      root: ~/.spack/opt/spack
      projections:
        all: "${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}"
      permissions:
        read: world
        write: user
  module_roots:  # <--- this is problematic because it's ambiguous: multiple install trees might all write to a singular tcl module root for example
    tcl:    $spack/share/spack/modules
    lmod:   $spack/share/spack/lmod
```